ppr-api/src/ppr_api/utils/validators/financing_validator.py

```diff
@@ -0,0 +1,198 @@
+VC_MH_NOT_ALLOWED = 'Vehicle Collateral type MH is not allowed with this registration type.\n'
+
+GC_NOT_ALLOWED_LIST = [MiscellaneousTypes.MH_NOTICE.value,
+                       PPSATypes.MARRIAGE_SEPARATION.value,
+                       PPSATypes.REPAIRER_LIEN.value,
+                       PPSATypes.MH_LIEN.value,
+                       PPSATypes.LAND_TAX.value]
+GC_ONLY_LIST = [PPSATypes.FORESTRY_CHARGE.value,
+                PPSATypes.FORESTRY_LIEN.value,
+                PPSATypes.FORESTRY_SUB_CHARGE.value,
+                MiscellaneousTypes.HC_NOTICE.value,
+                MiscellaneousTypes.WAGES_UNPAID.value]
+VC_REQUIRED_LIST = [MiscellaneousTypes.MH_NOTICE.value,
+                    PPSATypes.MARRIAGE_SEPARATION.value,
+                    PPSATypes.REPAIRER_LIEN.value,
+                    PPSATypes.MH_LIEN.value,
+                    PPSATypes.LAND_TAX.value]
+VC_MH_ONLY_LIST = [MiscellaneousTypes.MH_NOTICE.value,
+                   PPSATypes.MARRIAGE_SEPARATION.value,
+                   PPSATypes.MH_LIEN.value,
+                   PPSATypes.LAND_TAX.value]
+LIFE_INFINITE_LIST = [PPSATypes.MARRIAGE_SEPARATION.value,
+                      PPSATypes.MH_LIEN.value,
+                      PPSATypes.LAND_TAX.value]
+
+
+def validate(json_data):
+    """Apply validation rules for all financing statement registration types."""
+    error_msg = ''
+    try:
+        if 'type' not in json_data:
+            return error_msg
+
+        reg_type = json_data['type']
+        reg_class = get_registration_class(reg_type)
+        if reg_class is None:
+            return error_msg
+
+        error_msg += validate_life(json_data, reg_type, reg_class)
+        error_msg += validate_vehicle_collateral(json_data, reg_type, reg_class)
+        error_msg += validate_general_collateral(json_data, reg_type, reg_class)
+        error_msg += validate_trust_indenture(json_data, reg_type)
+        error_msg += validate_rl(json_data, reg_type)
+        error_msg += validate_other_description(json_data, reg_type)
+        return error_msg
+    except ValueError:
+        return error_msg
+
+
+def validate_life(json_data, reg_type: str, reg_class: str):
+    """Validate lifeYears and lifeInfinite by registration type."""
+    error_msg = ''
+    if reg_type in LIFE_INFINITE_LIST or reg_class in (model_utils.REG_CLASS_CROWN, model_utils.REG_CLASS_MISC):
+        if 'lifeYears' in json_data and (json_data['lifeYears'] != model_utils.LIFE_INFINITE):
+            error_msg = LY_NOT_ALLOWED
+        if 'lifeInfinite' in json_data and not json_data['lifeInfinite']:
+            error_msg += LI_INVALID
+    elif reg_type == model_utils.REG_TYPE_REPAIRER_LIEN and 'lifeInfinite' in json_data and json_data['lifeInfinite']:
+        error_msg += LI_NOT_ALLOWED
+    elif 'lifeYears' not in json_data and 'lifeInfinite' not in json_data:
+        error_msg += LIFE_MISSING
+    elif 'lifeYears' in json_data and 'lifeInfinite' in json_data and \
```

Comment:
```suggestion
    elif json_data.get('lifeInfinite') and json_data.get('lifeYears', -1) > 0
```
wont throw exceptions for missing values

Reply:
Suggested change made.